R/logging.R

```diff
@@ -0,0 +1,110 @@
+#' Get node log files from Azure Storage. By default, this operation will print the files on screen.
+#'
+#' @param cluster The cluster object
+#' @param nodeId Id of the node
+#' @param type The type of logs the user wants: "stdout" or "stderr"
+#' @param verbose Flag for printing log files onto console
+#'
+#' @param ... Further named parameters
+#' \itemize{
+#'  \item{"localPath"}: { Path to save file to }
+#'  \item{"overwrite"}: { Will only overwrite existing localPath }
+#'}
+#'
+#' @examples
+#' stdoutText <- getClusterLogs(cluster, "tvm-1170471534_1-20170829t072146z", type = "stdout", verbose = FALSE)
+#' getClusterLogs(cluster, "tvm-1170471534_2-20170829t072146z", type = "stderr", localPath = "abc.txt", overwrite = TRUE)
+#' @export
+getClusterLogs <-
+  function(cluster,
+           nodeId,
+           type = c("stdout", "stderr"),
+           verbose = TRUE,
+           overwrite = FALSE,
+           localPath = NULL) {
+    filePathString <- "startup/%s.txt"
+
+    filePath <- switch(type,
+                       "stdout" = {
+                         sprintf(filePathString, type)
+                       },
+                       "stderr" = {
+                         sprintf(filePathString, type)
+                       },
+                       stop(sprintf("Incorrect File Path Type: %s", type)))
+
+    nodeFileContent <- rAzureBatch::getNodeFile(
+      cluster$poolId,
+      nodeId,
+      filePath,
+      content = "text",
+      write = write,
+      progress = TRUE,
+      localPath = localPath,
+      overwrite = overwrite
+    )
+
+    if (verbose) {
+      cat(nodeFileContent)
+    }
+
+    nodeFileContent
+  }
+
+#' Get job log files from Azure Storage. By default, this operation will print the files on screen.
```

Comment:
This gets a bit more tricky since the nodes only write out the files to storage after the task is complete. We should probably document that.

Is it worth having some kind of support to pull task logs from the node itself? (scenario is that it is a long running task and users want to cat the log to see where its at)